### DISCUSSION TITLE
13vαにおいての計画

### DISCUSSION THREAD

**Discussion by NePonpon** [TUSB/TheUnusualSkyBlock#65] [Q&A]
**各項目にチェックボタンが配置してあるので、完成後チェックを入れる。**

## 目次
* [Player](#player)
  * [職業](#player-1)
  * [潜在能力](#player-2)
* [Item](#item)
  * [LootTable](#item-1)
  * [報酬チェスト](#item-2)
  * [通常アイテム](#item-3)
  * [特殊アイテム](#item-4)
    * [通貨](#item-5)
    * [魔石](#item-6)
    * [TUSBメモリー](#item-7)
    * [ポーション](#item-8)
  * [交易](#item-9)
* [Entity](#entity)
  * [Mob](#entity-1)
  * [Spawner](#entity-2)
* [建築](#architecture)
* [クリア条件](#clear)

<a id="player"></a>
# Player
プレイヤーの仕様

<a id="player-1"></a>
## 職業
現在進捗がよく進んでいる4職業を選択できるようにする。

- [x] 剣士
- [x] 忍者
- [x] 狩人
- [x] 召喚士

<a id="player-2"></a>
## 潜在能力
Lv50以降で手に入るCPを消費して成長させることができる潜在能力。

- [x] オーラ継続時間延長
CP1 = 2%
- [x] 消費MP軽減  
CP1 = 1%
- [x] スキル使用不可時間短縮
CP1 = 1%
- [x] 免疫力最低値
CP1 = 2%
- [x] 全属性攻撃の追加
CP1 = 1%
- [x] 他職スキル上限解放
CP2 = 1LvUP
- [x] 他ジョブLvUP
CP1 = 選択ジョブ1LvUP
- [x] CPリセット
CP0
 ※CPを使用前に初期化する際、一定時間CP変更を不可。

- [ ] ~~属性近接攻撃時の範囲増加~~
~~実装面倒~~
- [ ] ~~基礎魔法防御力の増加~~
~~CP1 = 1%~~ ※没

<a id="item"></a>
# Item
アイテム全般の仕様

<a id="item-1"></a>
## LootTable
- [x] Mobを倒しと通常世界の通貨とランダムな魔石(Lv1)を落とす。7:3ぐらいの割合？

<a id="item-2"></a>
## 報酬チェスト
- [x] TUSBメモリー及び今まで作られている盾や特殊武器が出る。

<a id="item-3"></a>
## 通常アイテム
羊毛とピッケル6種、各職業が使うアイテムのみ。(建築素材などはない)

## 宝石ブロックと宝石の仕様
宝石ブロックのレシピは消さず、そのままに(エメラルドブロックもレシピは消さない)
宝石ブロックでビーコンの起動ができないように調整。
エメラルドは通貨として使われているが、あまりにインフレするようであれば後で修正。

<a id="item-4"></a>
## 特殊アイテム
主にTUSBオリジナルアイテムの仕様

<a id="item-5"></a>
### 通貨

- [x] 通常世界の通貨のみ

<a id="item-6"></a>
### 魔石
全ての魔石を使用可能。  
- [ ] テクスチャーが未完成

<a id="item-7"></a>
### TUSBメモリー

- [x] 必要最低限の攻略で必須であるものを使用する。

<a id="item-8"></a>
### ポーション

- [x] マイクラのポーションは使用せず、TUSBメモリーとして登場させる

<a id="item-9"></a>
## 交易

- [x] 公益施設は飛空船内に設置される予定
- [x] エリアスポーンの行商人からしか入手できないアイテムもあり

- [x] 両替
  - [x] 羊毛32 → 通常世界の通貨1
  - [ ] 魔石1 → 通常世界の通貨10

- [x] アイテムガチャができるTUSBメモリー
使用するとランダムなTUSBメモリーが出現する。
  - [ ] 一部のTUSBメモリーはエリアスポーンの行商人でしか入手不可
  - [ ] 値段は相談。

- [x] ピッケル6種
  - [x] 木、石、鉄、金の4種類を取引可。
  - [ ] ダイヤとネザライトは行商人のみ。
  - [ ] 値段は相談。

- [x] 各職業が使うアイテム
人参棒、剣、斧、弓が取引可能。
  - [x] 値段は相談。

- [x] 防具
全ての防具及び通常の盾を取引可。
  - [ ] 値段は相談。

- [x] 魔石
  - [x] Lv4以下の魔石を取引可。
  - [ ] Lv6のランダムな魔石は行商人のみ。
  - [ ] 値段は相談。

<a id="entity"></a>
# Entity
エンティティの仕様

<a id="entity-1"></a>
## Mob

- [x] 全50体の改造Mob
現在数40弱。(弾や演出はこの中の数に含めない)

- [x] 羊毛
4、8、16、32、64、128の羊毛を落とすHP1の羊。  
金と銀、虹等はまだ作らない予定。
  - [ ] 羊毛施設は飛空船内にて設置予定。

<a id="entity-2"></a>
## Spawner

- [x] 功績値
邪悪なスポナーに限り、完全破壊した際に得られるポイント  
功績値 = 経験値 になるため、 功績値25の場合、経験値25を得ることができる。
  - [ ] クエスト屋の常時クエストで功績値を経験値に変換可能。

<a id="architecture"></a>
# 建築
以下の3つのみを使用

- [x] [以前例の企画で製作した島](https://youtu.be/iq0xqFSmcmk)
- [x] 飛空島にある飛空船
- [x] 職業島

<a id="clear"></a>
# クリア条件

- [ ] 13vαにおいてのクリア条件
島にある合計64個のすべてのスポナーを破壊し、攻略タイムを競う。

**Comment by anminmakura** [comment]
### v13αの初期地点について
クラウディア初期地点の完成が遅くなりそうであれば、
スポーン => 職業島 => v13α島

クラウディア初期地点に入れられるのであれば、
スポーン => クラウディア初期地点 => 職業島 => v13α島

**Comment by NePonpon** [comment]
潜在能力の「基礎魔法防御力の増加」を削除、没とした。

> Ponpon
> CPを割り振ることができるときにはすでにエンチャントなんて標準だろうから、エンチャントによる軽減とCPによる軽減が被るなら振らなさそう

**Comment by NePonpon** [comment]
# デバッグに便利なコマンド

## ◆一括設置◆

* 固定村人
`/function settings:alpha_debug/villager`

* スポナー
`/function settings:alpha_debug/spawner`

* 報酬チェスト等
`/function settings:alpha_debug/container`


## 一括消し
```
/kill @e[tag=Spawner,distance=..128]
/kill @e[type=villager,distance=..128]
/kill @e[tag=Mob,distance=..128]
```

## 経験値ボーナスリセット
※全員のプレイヤーデータを初期化した状態でこれを行ってください。
```
scoreboard players set $World AllExp 0
```

**Comment by NePonpon** [comment]
v13αは末に配布済みなのでClose
(激遅対応)
